### ISSUE TITLE
Serverless functions can't be uploaded through CLI

### ISSUE THREAD

**Issue by matiasvad** [HubSpot/hubspot-cms-tools#372]
Serverless function folders fail to upload through CLI.

I use CLI 2.1.0.

### Steps to reproduce
Create a new developer portal.
Connect with `hs init`
Follow Step 1 through 3 on your own guide: https://developers.hubspot.com/docs/cms/guides/getting-started-with-serverless-functions

Once `serverless.json` is saved, which triggers the upload, the CLI will return the following error:
```
[ERROR] Uploading file /Users/matias/test/serverless-tutorial/my-first-function.functions/serverless.json to serverless-tutorial/my-first-function.functions/serverless.json failed
[ERROR] Folder name cannot end with .functions
```

**Comment by drewjenkins** [comment]
Hey @matiasvad sorry you are encountering this.  The error you are seeing is actually a bit misleading.  What it is actually indicating is your account doesn't have access to serverless functions.  To have access to them, you must either be an enterprise tier customer, or on a [CSM Developer Sandbox Account](https://app.hubspot.com/signup/standalone-cms-developer?userType=developer&hubs_signup-url=developers.hubspot.com/docs/cms/guides/getting-started-with-serverless-functions&hubs_signup-cta=null) as suggested in the getting started guide.

If this is not the case, let me know and we can dig in further.  We have an issue open on our side to make that messaging more clear about the issue you are experiencing.

**Comment by matiasvad** [comment]
As I mentioned in the reproduction steps this happens when spinning up a

new Developer Portal.

I just tried it in a completely clean one.

I had the issue originally in a dev portal I spun up 14 days ago, for a

client project, and was debugging the issue.

On October 23, 2020, GitHub <notifications@github.com> wrote:
> Hey @matiasvad <https://github.com/matiasvad> sorry you are
> encountering this. The error you are seeing is actually a bit
> misleading. What it is actually indicating is your account doesn't
> have access to serverless functions. To have access to them, you must

> either be an enterprise tier customer, or on a CSM Developer Sandbox

> Account <https://app.hubspot.com/signup/standalone-cms-
> developer?userType=developer&hubs_signup-
> url=developers.hubspot.com/docs/cms/guides/getting-started-with-
> serverless-functions&hubs_signup-cta=null> as suggested in the getting
> started guide.
>
> If this is not the case, let me know and we can dig in further. We
> have an issue open on our side to make that messaging more clear about

> the issue you are experiencing.
>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/HubSpot/hubspot-cms-tools/issues/372#issuecomment-

715557353>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/AAOXDCNZT5TK7XPY5UQTA6DSMHM33ANCNFSM4S46JDHA>.

**Comment by matiasvad** [comment]
After a bit of searching around I figured the issue.

We're using an App Developer Account, to manage a bunch of our developer accounts and are using those for testing websites and other functionalities. These developer account can *not* access Serverless Functions. However, everything else is available.

When I created a specific CMS Developer Sandbox account I was able to access it.

The CLI is working as intended, it's just a matter of having the right type of dev account.
